influxdb3/management_decicated.go

```diff
@@ -0,0 +1,139 @@
+/*
+ The MIT License
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+package influxdb3
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+type (
+	// Dedicatedclient represents a client for InfluxDB Cloud Dedicated administration operations.
+	DedicatedClient struct {
+		client *Client
+	}
+
+	DedicatedClientConfig struct {
+		AccountID        string
+		ClusterID        string
+		ManagementToken  string
+		ManagementAPIURL url.URL
+	}
+
+	Database struct {
+		Name               string              `json:"name"`
+		MaxTables          uint64              `json:"maxTables"`          // default 500
+		MaxColumnsPerTable uint64              `json:"maxColumnsPerTable"` // default 250
+		RetentionPeriod    uint64              `json:"retentionPeriod"`    // nanoseconds default 0 is infinite
+		PartitionTemplate  []PartitionTemplate `json:"partitionTemplate"`  // Tag or TagBucket, limit is total of 7
+	}
+
+	PartitionTemplate interface {
+		IsPartitionTemplate()
+	}
+
+	Tag struct {
+		Type  string `json:"type"`
+		Value string `json:"value"`
+	}
+
+	TagBucket struct {
+		Type  string         `json:"type"`
+		Value TagBucketValue `json:"value"`
+	}
+
+	TagBucketValue struct {
+		TagName         string `json:"tagName"`
+		NumberOfBuckets uint64 `json:"numberOfBuckets"`
+	}
+)
+
+func (t Tag) IsPartitionTemplate()        {}
+func (tb TagBucket) IsPartitionTemplate() {}
+
+// NewCloudDedicatedClient creates new DedicatedClient with given InfluxDB client.
+func NewCloudDedicatedClient(client *Client) *DedicatedClient {
+	return &DedicatedClient{client: client}
```

Comment:
The default domain `console.influxdata.com` should be set here, or the example file should show that users must pass it.

Reply:
Thanks, I forgot to add this into the example

Reply:
Fixed in [f8c58b5](https://github.com/InfluxCommunity/influxdb3-go/pull/87/commits/f8c58b5d83b7b04e9f603d176105e549321c1030) and [53a8696](https://github.com/InfluxCommunity/influxdb3-go/pull/87/commits/53a8696dc89b716e0c16117b60cf7e2d69e45834)